File: src/com/themaskedcrusader/bukkit/config/SaveFile.java
/*
 * Copyright 2013 Topher Donovan (themaskedcrusader.com)
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package com.themaskedcrusader.bukkit.config;

import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.plugin.java.JavaPlugin;

public class SaveFile {
    private ConfigAccessor saveFile;

    public  SaveFile(JavaPlugin plugin, String fileName) {
        saveFile = new ConfigAccessor(plugin, fileName);
//        saveFile.reloadConfig();
//        saveFile.saveDefaultConfig();
    }

    public FileConfiguration getConfig() {
        return saveFile.getConfig();
    }

    public ConfigAccessor get() {
        return saveFile;
    }

    public void saveConfig() {
        saveFile.saveConfig();
    }

    public void delete() {
        saveFile.delete();
    }
}

File: src/main/kotlin/me/schalk/spigot/lib/config/SaveFile.kt
/*
 * Copyright (c) 2013 - 2023 Christopher Schalk
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package me.schalk.spigot.lib.config

import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.plugin.java.JavaPlugin

class SaveFile(plugin: JavaPlugin, fileName: String) {
    private var saveFile: ConfigAccessor

    init {
        saveFile = ConfigAccessor(plugin, fileName)
    }

    fun getConfig(): FileConfiguration {
        return saveFile.config
    }

    fun get(): ConfigAccessor {
        return saveFile
    }

    fun saveConfig() {
        saveFile.saveConfig()
    }

    fun delete() {
        saveFile.delete()
    }
}

